subgraphs/_reference_/src/sdk/protocols/perpfutures/pool.ts

```diff
@@ -0,0 +1,810 @@
+    amountsUSD = sortArrayByReference(newOrder, tokens, amountsUSD);
+
+    this.pool.rewardTokens = tokens;
+    this.pool.rewardTokenEmissionsAmount = amounts;
+    this.pool.rewardTokenEmissionsUSD = amountsUSD;
+    this.save();
+  }
+
+  /**
+   * Adds some value to the cumulativeUniqueBorrowers counter. If the value is omitted it will default to 1.
+   * If you are loading accounts with the AccountManager you won't need to use this method.
+   * @param count {u8} The value to add to the counter.
+   */
+  addBorrower(count: u8 = 1): void {
+    this.pool.cumulativeUniqueBorrowers += count;
+    this.save();
+  }
+
+  /**
+   * Adds some value to the cumulativeUniqueLiquidators counter. If the value is omitted it will default to 1.
+   * If you are loading accounts with the AccountManager you won't need to use this method.
+   * @param count {u8} The value to add to the counter.
+   */
+  addLiquidator(count: u8 = 1): void {
+    this.pool.cumulativeUniqueLiquidators += count;
+    this.save();
+  }
+
+  /**
+   * Adds some value to the cumulativeUniqueLiquidatees counter. If the value is omitted it will default to 1.
+   * If you are loading accounts with the AccountManager you won't need to use this method.
+   * @param count {u8} The value to add to the counter.
+   */
+  addLiquidatee(count: u8 = 1): void {
+    this.pool.cumulativeUniqueLiquidatees += count;
+    this.save();
+  }
+
+  /**
+   * Adds 1 to the cumulativePositionCount counter and adds 1 to the counter corresponding the given position type.
+   * If you are creating transaction entities from the Account class you won't need to use this method.
+   * @param positionSide {PositionType} The type of transaction to add.
+   * @see PositionType
+   * @see Account
+   */
+  openPosition(positionSide: PositionType): void {
+    if (positionSide == PositionType.LONG) {
+      this.pool.longPositionCount += 1;
+    } else if (positionSide == PositionType.SHORT) {
+      this.pool.shortPositionCount += 1;
+    }
+
+    this.pool.openPositionCount += 1;
+    this.pool.cumulativePositionCount += 1;
+    this.save();
+
+    this.protocol.addPosition(positionSide);
+  }
+
+  /**
+   * Subtracts 1 to the cumulativePositionCount counter and adds 1 to the counter corresponding the given position type.
+   * If you are creating transaction entities from the Account class you won't need to use this method.
+   * @param positionSide {PositionType} The type of transaction to add.
+   * @see PositionType
+   * @see Account
+   */
+  closePosition(positionSide: PositionType): void {
+    if (positionSide == PositionType.LONG) {
+      this.pool.longPositionCount -= 1;
+    } else if (positionSide == PositionType.SHORT) {
+      this.pool.shortPositionCount -= 1;
+    }
+
+    this.pool.openPositionCount -= 1;
+    this.pool.closedPositionCount += 1;
+    this.save();
+
+    this.protocol.closePosition(positionSide);
+  }
+
+  addVolumeByToken(tokenAddress: Address, amount: BigInt): void {
```

Comment:
The following 4 functions basically all do the same logic, you could make a generalizable function to handle these updates

Reply:
How do you suggest generalizing these 4 functions?

Reply:
You have a helper function that takes the tokenAddress and amount, then also pass in the type of add. And you have an if block or switch statement for the actual add block. 

The rest is the same for each function

Reply:
Feel free to resolve if you think this is unnecessary @harsh9200